frontend/src/pages/projects/screens/spawner/dataConnection/useNotebookDataConnection.ts

```diff
@@ -15,36 +20,33 @@ export const getNotebookDataConnection = (
 };
 
 export const useNotebookDataConnection = (
+  dataConnections: DataConnection[],
   notebook?: NotebookKind,
-  dataConnections: DataConnection[] = [],
 ): [
   dataConnection: DataConnectionData,
-  setDataConnection: (connection: DataConnectionData) => void,
+  setDataConnection: UpdateObjectAtPropAndValue<DataConnectionData>,
+  resetDefaults: () => void,
 ] => {
-  const [dataConnection, setDataConnection] = React.useState<DataConnectionData>({
+  const createDataState = useGenericObjectState<DataConnectionData>({
     type: 'creating',
     enabled: false,
   });
```

Comment:
Would it not make sense to set the default values and then remove the `useEffect` ?

```suggestion
  const notebookDataConnectionSecret = getNotebookDataConnection(notebook, dataConnections)?.data
    .metadata.name;
  const createDataState = useGenericObjectState<DataConnectionData>(
    notebookDataConnectionSecret
      ? {
          type: 'existing',
          enabled: true,
          existing: {
            secretRef: {
              name: notebookDataConnectionSecret,
            },
          },
        }
      : {
          type: 'creating',
          enabled: false,
        },
  );
```

Reply:
Wow, you are correct, didn't know why I used that way before...